src/Symfony/Component/Validator/Tests/Constraints/HexaColorValidatorTest.php

```diff
@@ -0,0 +1,228 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Tests\Constraints;
+
+use Symfony\Component\Validator\Constraints\HexaColor;
+use Symfony\Component\Validator\Constraints\HexaColorValidator;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+use Symfony\Component\Validator\Test\ConstraintValidatorTestCase;
+
+class HexaColorValidatorTest extends ConstraintValidatorTestCase
+{
+    protected function createValidator(): HexaColorValidator
+    {
+        return new HexaColorValidator(HexaColor::VALIDATION_MODE_BOTH);
+    }
+
+    public function testUnknownDefaultModeTriggerException()
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionMessage('The "defaultMode" parameter value is not valid.');
+        new HexaColorValidator('Unknown Mode');
+    }
+
+    public function testNullIsValid()
+    {
+        $this->validator->validate(null, new HexaColor());
+
+        $this->assertNoViolation();
+    }
+
+    public function testEmptyStringIsValid()
+    {
+        $this->validator->validate('', new HexaColor());
+
+        $this->assertNoViolation();
+    }
+
+    public function testObjectEmptyStringIsValid()
+    {
+        $this->validator->validate(new EmptyHexaColorObject(), new HexaColor());
+
+        $this->assertNoViolation();
+    }
+
+    public function testExpectsStringCompatibleType()
+    {
+        $this->expectException(UnexpectedValueException::class);
+        $this->validator->validate(new \stdClass(), new HexaColor());
+    }
+
+    /**
+     * @dataProvider getValidHexaColors
+     */
+    public function testValidHexaColors($hexaColor)
+    {
+        $this->validator->validate($hexaColor, new HexaColor());
+
+        $this->assertNoViolation();
+    }
+
+    public function getValidHexaColors()
+    {
+        return [
+            ['#ABCDEF'],
+            ['#abcdef'],
+            ['#C0FFEE'],
+            ['#c0ffee'],
+            ['#501311'],
+            ['#FAB'],
+            ['#fab'],
+            ['#F4B'],
+            ['#f4b'],
+            ['#907'],
+        ];
+    }
+
+    /**
+     * @dataProvider getValidHexaColorsHtml5
+     */
+    public function testValidHexaColorsHtml5($hexaColor)
+    {
+        $this->validator->validate($hexaColor, new HexaColor(['mode' => HexaColor::VALIDATION_MODE_HTML5]));
+
+        $this->assertNoViolation();
+    }
+
+    public function getValidHexaColorsHtml5()
+    {
+        return [
+            ['#ABCDEF'],
+            ['#abcdef'],
+            ['#C0FFEE'],
+            ['#c0ffee'],
+            ['#501311'],
```

Comment:
@welcoMattic 

How do you feel about adding support for hexadecimal colors with transparency?

- https://developer.mozilla.org/en-US/docs/Web/CSS/color_value
- https://css-tricks.com/8-digit-hex-codes/